File: src/nomanssave/cr.java
package nomanssave;

import java.io.File;
import javax.swing.filechooser.FileFilter;

class cr extends FileFilter {
   // $FF: synthetic field
   final cp fM;

   cr(cp var1) {
      this.fM = var1;
   }

   public String getDescription() {
      return "Companion Export File";
   }

   public boolean accept(File var1) {
      if (var1.isDirectory()) {
         return !var1.isHidden();
      } else {
         return var1.getName().endsWith(".pet") || var1.getName().endsWith(".egg");
      }
   }
}

File: NMSSaveEditor/nomanssave/lower/cr.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Windows.Forms;
using System.Globalization;

namespace NMSSaveEditor
{

public class cr : FileFilter {
   // $FF: synthetic field
   public cp fM;

   public cr(cp var1) {
      this.fM = var1;
   }

   public string getDescription() {
      return "Companion Export File";
   }

   public bool accept(FileInfo var1) {
      if (var1.isDirectory()) {
         return !var1.isHidden();
      } else {
         return var1.Name.EndsWith(".pet") || var1.Name.EndsWith(".egg");
      }
   }
}

}
